irohad/ametsuchi/impl/storage_impl.hpp

```diff
@@ -22,39 +22,29 @@
 #include "ametsuchi/reconnection_strategy.hpp"
 #include "common/result.hpp"
 #include "interfaces/common_objects/common_objects_factory.hpp"
-#include "interfaces/iroha_internal/block_json_converter.hpp"
 #include "interfaces/permission_to_string.hpp"
 #include "logger/logger_fwd.hpp"
 #include "logger/logger_manager_fwd.hpp"
 
 namespace iroha {
   namespace ametsuchi {
-
-    class FlatFile;
-
-    struct ConnectionContext {
-      explicit ConnectionContext(std::unique_ptr<KeyValueStorage> block_store);
-
-      std::unique_ptr<KeyValueStorage> block_store;
-    };
-
     class StorageImpl : public Storage {
      protected:
-      static expected::Result<ConnectionContext, std::string> initConnections(
-          std::string block_store_dir, logger::LoggerPtr log);
+      static expected::Result<std::unique_ptr<BlockStorage>, std::string>
+      initConnections(
+          std::unique_ptr<BlockStorageFactory> persistent_block_storage_factory,
+          logger::LoggerPtr log);
 
      public:
       static expected::Result<std::shared_ptr<StorageImpl>, std::string> create(
-          std::string block_store_dir,
           const PostgresOptions &options,
-          PoolWrapper pool_wrapper,
+          std::shared_ptr<PoolWrapper> pool_wrapper,
           std::shared_ptr<shared_model::interface::CommonObjectsFactory>
               factory,
-          std::shared_ptr<shared_model::interface::BlockJsonConverter>
-              converter,
           std::shared_ptr<shared_model::interface::PermissionToString>
               perm_converter,
-          std::unique_ptr<BlockStorageFactory> block_storage_factory,
+          std::unique_ptr<BlockStorageFactory> temporary_block_storage_factory,
+          std::unique_ptr<BlockStorageFactory> persistent_block_storage_factory,
```

Comment:
Is `persistent_block_storage_factory` really required? I think instead we can pass `std::unique_ptr<BlockStorage> persistent_block_storage` as a constructor parameter since it is used once. Please check https://github.com/igor-egorov/iroha/commit/2a6757ae4954464d0a7c8ac7f2ee54cf2ae1dba8